src/net/NetworkClient.cpp

```diff
@@ -0,0 +1,73 @@
+#include "NetworkClient.h"
+
+NetworkClient::NetworkClient(std::string host) :
+    socket(io_service), host(host) {
+}
+
+NetworkClient::~NetworkClient() {
+    io_service.stop();
+    service_thread.join();
+}
+
+bool NetworkClient::init() {
+    tcp::resolver resolver(io_service);
+    tcp::resolver::query query(tcp::v4(), host, PORT);
+    tcp::resolver::iterator endpoint_iterator = resolver.resolve(query);
+    try {
+        boost::asio::connect(socket, endpoint_iterator);
+    }
+    catch (...) {
+        return false;
+    }
+    service_thread = boost::thread(boost::bind(&NetworkClient::run_service, this));
+    initialized = true;
+    return true;
+}
+
+bool NetworkClient::is_initialized() const {
+    return initialized;
+}
+
+void NetworkClient::send(std::string message) {
+    boost::asio::write(socket, boost::asio::buffer(message));
+}
+
+bool NetworkClient::has_messages() {
+    return !message_queue.empty();
+}
+
+std::string NetworkClient::pop_message() {
+    if (!has_messages())
+        throw std::logic_error("No messages");
```

Comment:
I'm not a huge fan of exceptions in C++, since they may introduce memory leaks (no garbage collection). Better to return a status boolean and take in a reference parameter to fill with data IMO.